indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -0,0 +1,298 @@
+    statusTracker.onTaskSubmitted(supervisorId, taskStatus);
+  }
+
+  private class SchedulerManager
+  {
+    private final String supervisorId;
+    private final String dataSource;
+    private final ClientSqlQuery spec;
+    private final CronSchedulerConfig cronSchedulerConfig;
+
+    private ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus status;
+
+    /**
+     * TODO: Note that the last task submitted per supervisor should eventually be persisted in the metadata store,
+     * along with any other scheduler state, so that the batch supervisor can recover gracefully and
+     * avoid missing task submissions during rolling restarts, etc.
+     */
+    private volatile DateTime lastTaskSubmittedTime;
+
+    private SchedulerManager(
+        final String supervisorId,
+        final String dataSource,
+        final CronSchedulerConfig cronSchedulerConfig,
+        final ClientSqlQuery spec
+    )
+    {
+      this.supervisorId = supervisorId;
+      this.dataSource = dataSource;
+      this.cronSchedulerConfig = cronSchedulerConfig;
+      this.spec = spec;
+    }
+
+    private synchronized void startScheduling()
+    {
+      if (jobsExecutor.isTerminated() || jobsExecutor.isShutdown()
+          || status == ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_SHUTDOWN) {
+        return;
+      }
+
+      status = ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_RUNNING;
+      final Duration timeUntilNextSubmission = getTimeUntilNextTaskSubmission();
+      if (timeUntilNextSubmission == null) {
+        log.info("No more tasks will be submitted for supervisor[%s].", supervisorId);
+        return;
+      }
+
+      jobsExecutor.schedule(
+          () -> {
+            try {
+              // Check status inside the runnable again before submitting any tasks
+              if (status == ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_SHUTDOWN) {
+                return;
+              }
+              lastTaskSubmittedTime = DateTimes.nowUtc();
+              submitSqlTask(supervisorId, spec);
+              emitMetric("batchSupervisor/tasks/submit/success", 1);
+            }
+            catch (Exception e) {
+              emitMetric("batchSupervisor/tasks/submit/failed", 1);
+              log.error(e, "Error submitting task for supervisor[%s]. Continuing schedule.", supervisorId);
+            }
+            startScheduling(); // Schedule the next task
+          },
+          timeUntilNextSubmission.getMillis(),
+          TimeUnit.MILLISECONDS
+      );
+    }
+
+    private synchronized void stopScheduling()
+    {
+      status = ScheduledBatchSupervisorSnapshot.BatchSupervisorStatus.SCHEDULER_SHUTDOWN;
+    }
+
+    private void emitMetric(final String metricName, final int value)
+    {
+      emitter.emit(
+          ServiceMetricEvent.builder()
+                            .setDimension("supervisorId", supervisorId)
```

Comment:
Probably redundant info since the supervisorId is always `prefix__datasource`.
If it is still possible for users to override that scheme, use dimension `DruidMetrics.ID` instead.
Then it would be easier to view this dimension in downstream viz applications without having to update their configs.